packages/assets-controllers/src/NftController.ts

```diff
@@ -222,41 +218,235 @@ const controllerName = 'NftController';
 /**
  * The external actions available to the {@link NftController}.
  */
-type AllowedActions =
+export type AllowedActions =
   | AddApprovalRequest
   | NetworkControllerGetNetworkClientByIdAction;
 
+export type AllowedEvents =
+  | PreferencesControllerStateChangeEvent
+  | NetworkControllerNetworkDidChangeEvent;
+
+export type NftControllerStateChangeEvent = ControllerStateChangeEvent<
+  typeof controllerName,
+  NftControllerState
+>;
+
+export type NftControllerEvents = NftControllerStateChangeEvent;
+
 /**
  * The messenger of the {@link NftController}.
  */
 export type NftControllerMessenger = RestrictedControllerMessenger<
   typeof controllerName,
   AllowedActions,
-  never,
+  NftControllerEvents | AllowedEvents,
   AllowedActions['type'],
-  never
+  AllowedEvents['type']
 >;
 
-export const getDefaultNftState = (): NftState => {
-  return {
-    allNftContracts: {},
-    allNfts: {},
-    ignoredNfts: [],
-  };
+export const defaultNftControllerState: NftControllerState = {
```

Comment:
Is there a reason you prefer this change? Seems like we have some controllers that use an object and others that use a function to generate the default state, and I'm not sure which is better. It does seem like using the function guarantees that the default state object won't be mutated.

Reply:
I am not sure which one is better but yes we have inconsistency in usage between controller for default state. worth discussing aligning this with the team